### PR TITLE
[WIP] formula_installer: Don't check devtools if pouring

### PR DESCRIPTION
**Pull request by woodruffw**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This *should* stop Homebrew from nagging users about their outdated Xcode installation during bottle-only installs. I'll be able to test it locally in a few days (my mac is currently packed away).

Some notes:
* This fixes the current behavior by moving the development tools check out of `perform_preinstall_checks` and into `FormulaInstaller#install`. This should be fine, although it does cause us to run the check *N + 1* times for 1 formula and  *N* dependencies.
* Some users may prefer this behavior, but it may not be beneficial in the long term: users *should* be nagged about the presence of unsupported Xcode installs, as doing so lessens the amount of weird setup triaging we have to do. Warning the user instead of totally failing the install might be a workable middle ground.

### CODE REVIEW

File: Library/Homebrew/formula_installer.rb [hunk 1]
```diff
@@ -219,6 +219,10 @@ def build_bottle_postinstall
   end
 
   def install
+    if DevelopmentTools.installed? && !pour_bottle?
```
Comment (ilovezfs) [inline Library/Homebrew/formula_installer.rb]: any appreciable difference in speed depending on the order of those conditions?

**Comment by ilovezfs** [comment]
We may also be able to skip the check if it's bottle unneeded.

File: Library/Homebrew/formula_installer.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/formula_installer.rb]: Yeh, the latter is likely to be a bit faster so probably preferable?

**Comment by woodruffw** [comment]
This should be good to go, pinging @ilovezfs and @MikeMcQuaid for final comments (if any).

**Comment by ilovezfs** [comment]
Thanks @woodruffw! 🏆

**Pull request merged**